cloud/pkg/synccontroller/objectsync.go

```diff
@@ -36,58 +36,59 @@ func (sctl *SyncController) manageObject(sync *v1alpha1.ObjectSync) {
 	ret, err := sctl.kubeclient.Resource(gvr).Namespace(sync.Namespace).Get(context.TODO(), sync.Spec.ObjectName, metav1.GetOptions{})
 	if err != nil {
 		klog.Errorf("failed to get obj(gvr:%v,namespace:%v,name:%v), %v", gvr, sync.Namespace, sync.Spec.ObjectName, err)
+		sendFailEvents(sync, err)
 		return
 	}
 
-	nodeName := getNodeName(sync.Name)
-	if ret != nil {
-		object, err = meta.Accessor(ret)
-		if err != nil {
-			return
-		}
+	object, err = meta.Accessor(ret)
+	if err != nil {
+		klog.Errorf("failed to get obj with meta (gvr:%v,namespace:%v,name:%v), %v", gvr, sync.Namespace, sync.Spec.ObjectName, err)
+		sendFailEvents(sync, err)
+		return
+	}
+	syncObjUID := getObjectUID(sync.Name)
+	if syncObjUID != string(object.GetUID()) {
+		err = apierrors.NewNotFound(schema.GroupResource{
+			Group:    "",
+			Resource: sync.Spec.ObjectKind,
+		}, sync.Spec.ObjectName)
+		sendFailEvents(sync, err)
+		return
+	}
+	sendEvents(sync, object.GetResourceVersion(), object)
+}
 
-		syncObjUID := getObjectUID(sync.Name)
-		if syncObjUID != string(object.GetUID()) {
-			err = apierrors.NewNotFound(schema.GroupResource{
-				Group:    "",
-				Resource: sync.Spec.ObjectKind,
-			}, sync.Spec.ObjectName)
-		}
+func sendFailEvents(sync *v1alpha1.ObjectSync, err error) {
+	if err == nil {
+		return
 	}
+	nodeName := getNodeName(sync.Name)
+	if apierrors.IsNotFound(err) {
+		// build default obj
+		obj := &unstructured.Unstructured{}
+		obj.SetName(sync.Spec.ObjectName)
+		obj.SetUID(types.UID(getObjectUID(sync.Name)))
+		obj.SetNamespace(sync.Namespace)
 
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			obj := &unstructured.Unstructured{}
-			obj.SetName(sync.Spec.ObjectName)
-			obj.SetUID(types.UID(getObjectUID(sync.Name)))
-			obj.SetNamespace(sync.Namespace)
-		} else {
-			klog.Errorf("Failed to manage pod sync of %s in namespace %s: %v", sync.Name, sync.Namespace, err)
-			return
-		}
-	}
-	sendEvents(err, nodeName, sync, sync.Spec.ObjectKind, object.GetResourceVersion(), object)
+		//trigger the delete event
+		klog.Infof("%s: %s has been deleted in K8s, send the delete event to edge", sync.Spec.ObjectKind, sync.Spec.ObjectName)
+		msg := buildEdgeControllerMessage(nodeName, sync.Namespace, sync.Spec.ObjectKind, sync.Spec.ObjectName, model.DeleteOperation, obj)
```

Comment:
cloudcore would crash when msg is nil, if you pod in edge use a pvc, and i found pvc's objectsync Spec.ObjectKind is empty, so buildEdgeControllerMessage return nil.
here, you should handle condition if msg is nil